js/client-mainmenu.js

```diff
@@ -54,8 +54,8 @@
 			buf += '<p><button class="button mainmenu3" name="joinRoom" value="ladder">Ladder</button></p></div>';
 
 			buf += '<div class="menugroup"><p><button class="button mainmenu4 onlineonly disabled" name="joinRoom" value="battles">Watch a battle</button></p>';
-			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p></div>';
-
+			buf += '<p><button class="button mainmenu5 onlineonly disabled" name="finduser">Find a user</button></p>';
+			buf += '<p class=link-generatore><button id="challeng-via-link" class="button mainmenu5 onlineonly">send challenge via link</button></p></div>';
```

Comment:
```suggestion
			buf += '<p class=link-generator><button class="button mainmenu5 onlineonly">Create challenge link</button></p></div>';
```
Button ids not preferred.